CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_CI_OPTC.uc

```diff
@@ -0,0 +1,124 @@
+//---------------------------------------------------------------------------------------
+//  AUTHOR:  statusNone
+//  PURPOSE: Modifies Templates in order to make infinite items single build instead
+//---------------------------------------------------------------------------------------
+//  WOTCStrategyOverhaul Team
+//---------------------------------------------------------------------------------------
+class X2Helper_CI_OPTC extends Object config (SingleBuildItems);
+
+struct SingleBuildItem
+{
+	var name SchematicName;
+	var int TradingPostValue;
+	
+	var array<StrategyCost> Costs;
+};
+
+var config array<SingleBuildItem> SingleBuildItems;
+
+static function ModifyTemplates()
+{
+	local array<X2DataTemplate> DifficultyVariants;
+	local X2ItemTemplateManager TemplateManager;
+	local X2SchematicTemplate Schematic;
+	local X2ItemTemplate Template;
+	local SingleBuildItem Item;
+	local int index, ValidIndex;
+
+	TemplateManager = class'X2ItemTemplateManager'.static.GetItemTemplateManager();
+	
+	foreach default.SingleBuildItems(Item)
+	{
+		Template = TemplateManager.FindItemTemplate(Item.SchematicName);
+		Schematic = X2SchematicTemplate(Template);
+
+		// look for a SchematicTemplate from the config array
+		if (Schematic != none)
+		{
+			// check if this schematic is flagged to require difficulty variants
+			// cycle through and modify each difficulty variant for this template
+			TemplateManager.FindDataTemplateAllDifficulties(Item.SchematicName, DifficultyVariants);
+			for (index = 0; index < DifficultyVariants.Length - 1; index++)
+			{
+				Schematic = X2SchematicTemplate(DifficultyVariants[index]);
+				
+				ValidIndex = index;
+				// if no cost value was set for this difficulty find the last useable value
+				while (ValidIndex > 0 && Item.Costs[ValidIndex].ResourceCosts.Length == 0)
+				{
+					ValidIndex--;
+				}
+				
+				Schematic.bSquadUpgrade = false;
+				Schematic.OnBuiltFn = BuildItem;
+				Schematic.HideIfPurchased = '';
+				Schematic.bOneTimeBuild = false;
+				Schematic.Cost = Item.Costs[ValidIndex];
+
+				// only modify item template varaints on first pass or if nessecary
+				if (Template.bShouldCreateDifficultyVariants || index == 0)
+				{
+					Template = TemplateManager.FindItemTemplate(Schematic.ReferenceItemTemplate);
```

Comment:
What is the point of this? Also, you don't actually handle the difficulty variants here

Reply:
that's intentional there is only a handful of items that actually have a matching ItemTemplate to go with the SchematicTemplate variant (other times the SchematicTemplate variants just point to the same ItemTemplate) so this just updates on first pass and then only if nessecary afterwards.. seemed silly to just update the same template 4 times when we didnt need to (updated the comment to better explain this)